web/src/test/stores/transaction.test.ts

```diff
@@ -0,0 +1,115 @@
+import { describe, test, expect } from 'vitest'
+
+describe('Transactions Store', () => {
+  enum TransactionType {
+    INCOME = "Income",
+    EXPENSE = "Expense",
+  }
+
+  const sampleTransaction = {
+    id: "Hi",
+    type: TransactionType.EXPENSE,
+    store: "Fortnite",
+    amount: 99,
+    date: new Date("2023-01-01"),
+    category: "Food",
+  }
+  const sampleIncome = {
+    id: "SuperHi",
+    type: TransactionType.INCOME,
+    store: "Bounty Hunting",
+    amount: 199,
+    date: new Date("2023-01-12"),
+    category: "Income",
+  }
+  const sampleSecondaryTransaction = {
+    id: "LowHi",
+    type: TransactionType.EXPENSE,
+    store: "Fortnite",
+    amount: 25,
+    date: new Date("2023-03-01"),
+    category: "Food",
+  }
+  test('Test is isEmpty', async () => {
+    const transactions = useTransactionStore()
+
+
+    expect(transactions.isEmpty).toBe(true);
+  })
+
+  test('Transaction can be added', async () => {
+    const transactions = useTransactionStore()
+
+    transactions.addTransaction(sampleTransaction)
+
+    expect(transactions.transactions.length).toBeGreaterThan(0);
+  })
+
+  test('Transactions have a Total Balance ', async () => {
+    const transactions = useTransactionStore()
+
+
+    expect(transactions.totalBalance).toBe(-99);
```

Comment:
It looks like these tests are depending on each other? They probably shouldn't do that. The tests should all be fully independent. You can setup a before each like in budget.test.ts that clears out all transactions for each test.

Reply:
I don't understand? It doesn't look like they are depending on one another?

Reply:
This test should fail as is with a totalBalance of 0, not -99. It is depending on a previous test to add a transaction to the store so that the totalBalance equals -99. Multiple other tests also have similar issues.

Reply:
From a good software perspective, we should not merge this code. From a class perspective where we want a passing grade, it might be worth just ignoring for now and fixing it in the next iteration...